Explorer/Assets/DCL/AvatarRendering/Wearables/Systems/ResolveAvatarAttachmentThumbnailSystem.cs

```diff
@@ -22,8 +24,32 @@ public partial class ResolveAvatarAttachmentThumbnailSystem : BaseUnityLoopSyste
         protected override void Update(float t)
         {
             CompleteWearableThumbnailDownloadQuery(World);
+            CompleteWearableABThumbnailDownloadQuery(World);
         }
 
+        [Query]
+        private void CompleteWearableABThumbnailDownload(in Entity entity, ref IAvatarAttachment wearable, ref AssetBundlePromise promise)
+        {
+            if (promise.LoadingIntention.CancellationTokenSource.IsCancellationRequested)
+            {
+                World.Destroy(entity);
+                return;
+            }
+
+            if (promise.TryConsume(World, out var result))
```

Comment:
To be honest, i find better to use `promise.SafeTryConsume` because if its already consumed somehow, this process will fail and will never be able to destroy the promise's entity leading into error spamming. Your call..

Reply:
Sure, why not

Reply:
Actually, I prefer to leave TryConsume. Logic is that over here it should be consumed only once, if not, I prefer to catch the exception